src/components/VideoEditor.tsx

```diff
@@ -0,0 +1,229 @@
+    const newVidData: Partial<Ambiance> = {
+      ...vidData,
+      [key]: event.target.value
+    }
+    setVidData(newVidData);
+  }
+
+  const handleOnChangeChannel = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, key: keyof Channel) => {
+    let newChannel: Channel;
+    if (key === "name") {
+      newChannel = {
+        link: vidData?.channel?.link || "",
+        name: event.currentTarget.value
+      }
+    } else {
+      newChannel = {
+        link: event.currentTarget.value,
+        name: vidData?.channel?.name || "",
+      }
+    }
+
+    const newVidData: Partial<Ambiance> = {
+      ...vidData,
+      channel: newChannel
+    }
+    setVidData(newVidData);
+  }
+
+
+  return (
+    <div id='Video-Editor'>
+      <Stack className='text-field-stack' direction="column" spacing={.4}>
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.name}
+          onChange={(e) => handleOnChange(e, "name")}
+          helperText="Display Name"
+          size='small'
+          variant="filled"
+          required
+          value={vidData?.name || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.group}
+          onChange={(e) => handleOnChange(e, "group")}
+          helperText="Subcategory"
+          variant="filled"
+          size='small'
+          required
+          value={vidData?.group || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.code}
+          onChange={(e) => handleOnChange(e, "code")}
+          helperText="YouTube URL"
+          variant="filled"
+          size='small'
+          required
+          value={vidData?.code || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.channel?.name}
+          onChange={(e) => handleOnChangeChannel(e, "name")}
+          helperText="Optional: YouTube channel Name"
+          variant="filled"
+          size='small'
+          value={vidData?.channel?.name || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.channel?.link}
+          onChange={(e) => handleOnChangeChannel(e, "link")}
+          helperText="Optional: YouTube channel Link"
+          variant="filled"
+          size='small'
+          value={vidData?.channel?.link || ""}
+        />
+        <TextField
+          sx={{ '& .dsoQSg': { paddingTop: '10px' } }}
+          defaultValue={vidData?.startTimeS}
+          onChange={(e) => handleOnChange(e, "startTimeS")}
+          helperText="Optional: Video start time (seconds)"
+          variant="filled"
+          size='small'
+          value={vidData?.startTimeS || ""}
+        />
```

Comment:
This field is causing a bug in DB as we are storing the startTimeS as a String and not an int. Will need to convert to int before saving to DB
![Screenshot 2024-04-05 at 10 49 11 AM](https://github.com/hermanator608/ambiance/assets/29784945/eabde58f-b871-4fcb-983f-3b8ab4b0b4d3)

Reply:
The reason TS is not yelling at us is because of out fancy single onChange handler bitting us in the but

https://github.com/hermanator608/ambiance/pull/11/files#diff-08ce9786c874e95b534e02708f4a28febe8b1f90f09876ce75d390a3857d4c16R74-R80

If we explicitly did { startTimeS: "12" } - TS would be like no way thats not an int

Reply:
Maybe also we change this to be an number type input to prevent letters
![Screenshot 2024-04-05 at 10 54 04 AM](https://github.com/hermanator608/ambiance/assets/29784945/0223e33b-afa3-45ce-aaf5-74ffb7145859)

Reply:
Allows us to then set a `min` as well so users dont type in negative numbers